test/Lachain.CryptoTest/ChainIdTest.cs

```diff
@@ -229,6 +229,8 @@ public bool VerifySignatureHashed(byte[] messageHash, byte[] signature, byte[] p
 
                 var parsedSig = new byte[65];
                 var recId = (RestoreEncodedRecIdFromSignatureBuffer(signature) - 36) / 2 / ChainId (useNewChainId);
+                if (recId < 0 || recId > 3)
```

Comment:
I think if we can add `recId >= 0 && recId <= 3` here as well it is much readable, Because we are throwing exception in different format and both of them means same